test/libsolidity/astPropertyTests/forLoop/isSimpleCounterLoop.sol

```diff
@@ -0,0 +1,180 @@
+            assembly {
+                x := i
+            }
+        }
+        /// BinaryOperationConditionRHS: isSimpleCounterLoop
+        for(uint i = 0; i < i + 1; i++) {
+        }
+        /// FreeFunctionConditionRHS: isSimpleCounterLoop
+        for(uint i = 0; i < h(); ++i) {
+        }
+        // Negative Cases
+        /// AdditionLoopExpression: isSimpleCounterLoop
+        for(uint i = 0; i < 42; i = i + 1) {
+        }
+        /// SimplePreDecrement: isSimpleCounterLoop
+        for(uint i = 42; i > 0; --i) {
+        }
+        /// SimplePosDecrement: isSimpleCounterLoop
+        for(uint i = 42; i > 0; i--) {
+        }
+        /// MultiplicationLoopExpression: isSimpleCounterLoop
+        for(uint i = 1; i < 42; i = i * 2) {
+        }
+        /// CounterIncrementLoopBody: isSimpleCounterLoop
+        for(uint i = 0; i < 42; ++i) {
+            i++;
+        }
+        /// CounterAssignmentLoopBody: isSimpleCounterLoop
+        for(uint i = 0; i < 42; ++i) {
+            i = 43;
+        }
+        /// CounterAssignmentInlineAssemblyLoopBody: isSimpleCounterLoop
+        for(uint i = 0; i < 42; ++i) {
+            assembly {
+                i := add(i, 1)
+            }
+        }
+        uint j = type(uint).max;
+        /// ExternalCounterLoopExpression: isSimpleCounterLoop
+        for (uint i = 0; i < 10; ++j) {
+        }
+        /// CounterIncrementRHSAssignment: isSimpleCounterLoop
+        for(uint i = 0; i < 10; ++i) {
+            x = i++;
+        }
+        /// NoEffectLoopExpression: isSimpleCounterLoop
+        for(uint i = 0; i < 42; i) {
+        }
+        /// EmptyLoopExpression: isSimpleCounterLoop
+        for(uint i = 0; i < 10; ) {
+        }
+        uint y = type(uint8).max + 1;
+        /// DifferentCommonTypeCondition: isSimpleCounterLoop
+        for(uint8 i = 0; i < y; ++i) {
+        }
+        /// LessThanOrEqualCondition: isSimpleCounterLoop
+        for(uint i = 0; i <= 10; ++i) {
+        }
+        /// ComplexExpressionCondition: isSimpleCounterLoop
+        for(uint i = 0; (i < 10 || g()); ++i) {
+        }
+        /// FreeFunctionConditionLHS: isSimpleCounterLoop
+        for(uint i = 0; h() < 100; ++i) {
+        }
+        /// FreeFunctionConditionDifferentCommonTypeLHS: isSimpleCounterLoop
+        for(uint8 i = 0; i < h(); ++i) {
+        }
+        /// NonIntegerTypeCondition: isSimpleCounterLoop
+        for(uint i = 0; address(this) < msg.sender; ++i) {
+        }
+        /// UDVTOperators: isSimpleCounterLoop
+        for(UINT i = UINT.wrap(0); i < UINT.wrap(10); i = i + UINT.wrap(1)) {
+        }
+        /// CounterAssignmentConditionRHS: isSimpleCounterLoop
+        for(uint i = 0; i < (i = i + 1); ++i) {
+        }
+        /// LiteralDifferentCommonTypeConditionRHS: isSimpleCounterLoop
+        for(uint8 i = 0; i < 257 ; ++i) {
+        }
+        /// StateVarCounterModifiedFunctionConditionRHS: isSimpleCounterLoop
+        for (z = 1; z < modifyStateVarZ(); ++z) {
```

Comment:
Needs fixing.
What would be the best approach here, regarding the rhs of the condition: (a) check forbidden/allowed cases or (b) restrict to only integer literals and variables?